src/deepsparse/loggers/metric_functions/natural_language_processing/token_classification/built_ins.py

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Set of functions for logging metrics from the token classification pipeline
+"""
+from typing import Dict, List
+
+import numpy
+
+
+__all__ = ["mean_score", "percent_zero_labels"]
+
+
+def percent_zero_labels(
+    token_classification_output: "TokenClassificationOutput",  # noqa: F821
+) -> Dict[str, float]:
+    """
+    Returns the percentage of zero labels in the token classification output
+
+    :param token_classification_output: the TokenClassificationOutput object
+    :return: A dictionary where the key is the token sequence index and the
+        value is the percentage of zero labels in the sequence of tokens
+    """
+    result = {}
+    for prediction_idx, prediction in enumerate(
```

Comment:
how would the enumerated values be logged currently? they would look like `percent_zero_labels__{idx}` right? thinking we'd prefer for each item in the batch to be logged separately to the same identifier, no? (this goes for all the fns we handle this way)

Reply:
Good point...

The user probably doesn't care about the specific ordering of their predictions in the dataset, so this information does seem to be adding clutter, not value. 
Ideally, in this case, we would like to have a dict like `{"": 10, "":11}` (so we get `percent_zero_labels_0.6` etc.) which obviously is not feasible.

Let's agree to land this PR for now, I will create a ticket for myself and handle the logging of items in a batch in a different PR ok?

Reply:
we need to think more here on how to handle batchwise VS item-wise logs. we might even want a wrapper object such so we can do something like:

```python
# wrapper class
class BatchLogResult:
    results: List[Any]

# example function
def item_wise_function(batch) -> BatchLogResult:
    return BatchLogResult([my_function(item) for item in batch])

# example leaf logger handling
class SomeLeafLogger:
    def log(item):
        if isinstance(item, BatchLogResult):
            # recurse for batch item in BatchLogResult.results
        else:
            self.logger.log(item)
```

Reply:
Agree